docs/xks/developer-guide/observability.md

```diff
@@ -0,0 +1,253 @@
+        }
+      ]
+```
+
+Check the official [Datadog Metrics Documentation](https://docs.datadoghq.com/agent/kubernetes/prometheus/) for more detailed information.
+
+### Tracing
+
+Datadog tracing is done with Application Performance Monitoring (APM), which sends traces from an application to Datadog.
+For traces to work the application needs to be configured with the language specific libraries.
+Check the [Language Documentation](https://docs.datadoghq.com/tracing/setup_overview/)
+for language specific instructions. Some of the languages that are supported are.
+
+- Golang
+- C#
+- Java
+- Python
+
+Configure your Deployment with the `DD_AGENT_HOST` environment for the APM agent to know where to send the traces.
+
+```deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  containers:
+    - env:
+        - name: DD_AGENT_HOST
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+```
+
+Check the official [Datadog Tracing Documentation](https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm) for more detailed information.
+
+### networkpolicy datadog
+
+When using XKF and your cluster have enabled datadog the tenant namespace will get a networkpolicy automatically that allows egress for tracing.
+
+You can view these rules by typing
+
+```shell
+kubectl get networkpolicies -n <tenant>
+```
+
+## Opentelemetry
+
+To gather opentelemtry data we rely on the [grafana agent operator](https://github.com/grafana/agent).
+The grafana agent operator deploys a grafana-agent in a central namespace configured as a part of XKF.
+
+The grafana agent gathers both metrics and logs and is able to receive traces.
+
+### Metrics
+
+To gather metrics data we use servicemonitors or podmonitors that is managed in XKF using the [prometheus-operator](https://github.com/prometheus-operator/prometheus-operator/).
+
+The prometheus-operator have great [getting started guide](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md)
+But if you want a quick example you can look below.
+
+The most important part for you as a developer you have to label the pod/service monitor yaml with `xkf.xenit.io/monitoring: tenant`,
+else the grafana agent won't find the rule to gather the metric.
+
+The selectors is used to find ether the pod or the service that you want to monitor.
+
+The podmonitor is used when you don't have a service in-front of your pod.
+For example this might be the case when your application don't use a HTTP endpoint to get requests.
+
+```podmonitor.yaml
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: podmonitor-example
+  labels:
+    xkf.xenit.io/monitoring: tenant
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: app1
+      app.kubernetes.io/instance: app1
+  podMetricsEndpoints:
+    - port: http-metrics
+```
+
+A servicemonitor is used to monitor a specific service.
```

Comment:
Use a servicemonitor when you when you have a service in-front of your pod ? Or do we have any other recommendations?